nano/test_common/testutil.hpp

```diff
@@ -63,6 +64,17 @@
 	}                                              \
 	EXPECT_NO_ERROR (ec);
 
+/*
+ * Asserts that the `val1 == val2` condition becomes true within the deadline
+ * Checked expressions should not modify any state
+ */
+#define ASSERT_TIMELY_EQ(time, val1, val2)         \
+	system.deadline_set (time);                    \
+	while (!((val1) == (val2)) && !system.poll ()) \
```

Comment:
This performs the equality check twice, if the first check is false and the second check true, it will fail to assert even though it should assert.

Reply:
That's why the documentation states expressions can't modify state. I found it very useful to debug ASSERT TIMELY failures quicker